.github/workflow_scripts/version_diff.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# List files in the specified S3 bucket and folder
+files=$(aws s3 ls s3://autogluon-ci/package_versions/)
+
+# Extract the filenames with the pattern package_versions_{datetimestamp}.txt
+latest_file=""
+latest_timestamp=0
+while read -r line; do
+    filename=$(echo $line | awk '{print $4}')
+    timestamp=$(echo $filename | cut -d'_' -f3 | cut -d'.' -f1)
+    if [ -n "$timestamp" ]; then
+        formatted_timestamp=$(echo $timestamp | sed 's/-/ /3')
+        # Convert to UNIX format
+        timestamp_seconds=$(date -d "$formatted_timestamp" +%s)
+    else
+        timestamp_seconds=0
+    fi
+    
+    if [ $timestamp_seconds -gt $latest_timestamp ]; then
+        latest_timestamp=$timestamp_seconds
+        latest_file=$filename
+    fi
+done <<< "$files"
+
+aws s3 cp s3://autogluon-ci/package_versions/$latest_file ./
+old_latest_file="old_${latest_file}"
+mv "./$latest_file" "./$old_latest_file"
+
+diff ./package_versions_* ./old_package_versions_* > ./diff_output.txt
```

Comment:
The diff command compares files line by line in a sequential manner. If the packages are listed in a different order between two files, diff will report these as differences, even if the same packages and versions are present. This can make the output noisy and harder to understand, especially in case of package addition or deletion

Reply:
`pip freeze` will always list the packages in a sorted order so the `diff` command will compare two sorted files.
If a new package is added or deleted then yes the output may be noisy, that's where we would download the current and previous file, remove the line containing the latest added or deleted package and then `diff`, this effort may have to be manual. It seems to be okay as we do not add or delete packages often

Reply:
The current implementation is acceptable as it primarily functions as a support tool to help authors determine if functional failures are associated with package version changes.